File: backend/src/RevisionPlatform.Api/Modules/Matching/MatchingContent.cs
namespace RevisionPlatform.Api.Modules.Matching;

/// <summary>
/// Content of a matching module: pairs of a concept and its definition. Learners
/// match each concept with a definition; optional instructions are shown above.
/// </summary>
public record MatchingContent(string? Instructions, List<MatchingPair?>? Pairs);

/// <summary>A concept and its definition. <see cref="Id"/> is unique within the module.</summary>
public record MatchingPair(string? Id, string? Concept, string? Definition);

File: frontend/src/app/modules/matching/matching-content.ts
/** Content of a matching module (see docs/api.md). */
export interface MatchingContent {
  instructions: string | null;
  pairs: MatchingPair[];
}

export interface MatchingPair {
  id: string;
  concept: string;
  definition: string;
}

export const MATCHING_LIMITS = {
  instructionsMaxLength: 500,
  minPairs: 2,
  maxPairs: 10,
  conceptMaxLength: 200,
  definitionMaxLength: 1000,
} as const;
